docs/CONTRIBUTING.md

```diff
@@ -19,33 +19,129 @@ Please submit pull requests in response to open issues. If you have a bug or fea
 
 ## Environment Setup
 
-**Prerequisites:**
+### Prerequisites:
 
 - [Node.js (npm)](https://nodejs.org/en/download/)
-- [Redis](https://redis.io/download)
+- [Redis](https://redis.io/) (2 methods)
+  - Use [Docker and docker-compose](#docker-and-docker-compose-set-up)
+  - Install as a [native application](#using-redis-as-a-native-application)
 
-**Redis Set Up**
-Some helpful guides:
+### Using Redis as a native application
 
-- [Redis for Linux](https://redis.io/download#installation) (recommended but not required)
-- [Redis for Windows Subsystem for Linux](https://anggo.ro/note/installing-redis-in-ubuntu-wsl/) (not recommended by Redis)
-- Redis for Windows - You can use [this](https://github.com/tporadowski/redis/releases) installer to get up and running quickly for now.
-- Redis for MacOS - need instructions
-  An easier solution would be to use Docker.
+#### Linux:
 
-**Setup**
+Install Redis using your distribution's package manager, for example:
+
+- Ubuntu based: `sudo apt install redis`
+- Red Hat, Fedora: `sudo dnf install redis`
+
+_Once Redis is installed, you can start it in a terminal by running:_
+
+```
+redis-server
+```
+
+#### Windows:
+
+We are using [Chocolatey package manager](https://chocolatey.org/) to install Redis on Windows. To get Chocolatey, simply follow this [guide](https://chocolatey.org/install) and run the following commands:
+
+1. To install Redis: `choco install redis-64 -v`
+1. To set Redis as a windows service: `redis-server --service-install`
+1. To start Redis: `redis-server --service-start`
+1. To check if running and display server information: `redis-cli info`
+
+### Docker and Docker-Compose Set Up
+
+#### Windows 10 Pro, Enterprise, or Education
+
+1. Get [Docker for Desktop For Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows)
+1. Docker for Desktop comes with docker-compose installed.
+
+#### MacOS (Sierra 10.12 or above)
+
+1. Get [Docker for Desktop For Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac)
+1. Docker for Desktop comes with docker-compose installed.
+
+#### Windows 10 Home Edition and Windows Subsystem for Linux (WSL)
+
+Docker Desktop for Windows is not available on Home Edition, and you cannot run docker in WSL (Windows Subsystem for Linux). You can get the environment set up using the following methods:
+
+- Get [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/) instead of Docker Desktop for Windows. Make sure that your system can do virtualization and enable virtualization.
+- Set up a virtual machine to run Linux Ubuntu
+  1. [Download VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+  1. [Follow this helpful youtube tutorial to create a virtual machine with Ubuntu](https://www.youtube.com/watch?v=ThsxqznrgCw&t=401s)
+  1. Use the Linux installation instructions [below](#linux-Ubuntu).
+- Use Windows 10 Education Edition
```

Comment:
Awesome work! One think to consider might be adding a few words to help emphasize the fact that users need only choose _one_ of the three listed setups. This would help ensure that users do not accidentally mistake these the unordered list items presented here as representing individual steps all needed to be performed to set up the environment.

e.g. you might "You can get the environment set up using the following methods" with something like "You can get the environment set up using **_one_** of the three following methods".

Reply:
Very true, thanks Rafi, I will get that done asap.